### DISCUSSION TITLE
Azureにログを出力する方法を知りたい

### DISCUSSION THREAD

**Discussion by kharada7** [AlesInfiny/maris#1848] [05: Q&A] [answered]
## 概要

AlesInfiny Marisの利用プロジェクトでAzureにログを出力するための実装方法が現状さっぱり分からないので、要件別サンプルにあると嬉しい。

## 完了条件

Azureにログを出力できる要件別サンプルが追加されること。

**Comment by KentaHizume** [comment]
Application Insightsへログ出力できることは以前確認した形跡がございましたので、
ブランチが残っていれば活用できるかもしれません。

- https://github.com/AlesInfiny/maris/issues/814#issuecomment-2031278427

**Comment by tsuna-can-se** [comment, marked as answer]
AlesInfiny MarisではILoggerを利用してログ出力を行います。
Application Isightsへのログ出力は、[Microsoft.Extensions.Logging.ApplicationInsights](https://www.nuget.org/packages/Microsoft.Extensions.Logging.ApplicationInsights)のパッケージを利用します。
エントリーポイントのプロジェクトの参照に追加して、Program.csとappsettings.jsonに設定を実装/記述することとなります。
Webアプリケーションとコンソールアプリケーションで設定方法は一部異なります。
設定方法、実装方法の詳細は以下に解説がありますのでご確認ください。
<https://learn.microsoft.com/ja-jp/azure/azure-monitor/app/ilogger>
